packages/smooth_app/lib/pages/offline_tasks_page.dart

```diff
@@ -0,0 +1,160 @@
+        ),
+      ),
+      body: RefreshIndicator(
+        onRefresh: () async {
+          setState(() {});
+        },
+        child: Center(
+          child: FutureBuilder<List<Task>>(
+            future: _fetchListItems(),
+            builder:
+                (BuildContext context, AsyncSnapshot<List<Task>> snapshot) {
+              if (snapshot.connectionState == ConnectionState.waiting) {
+                return const Center(
+                  child: CircularProgressIndicator(),
+                );
+              }
+              if (snapshot.data!.isEmpty) {
+                return const Center(child: Text('No Pending Tasks'));
+              } else {
+                return ListView.builder(
+                  itemCount: snapshot.data!.length,
+                  itemBuilder: (BuildContext context, int index) {
+                    if (snapshot.data!.isEmpty) {
+                      return const Center(
+                        child: Text(
+                          'No data',
+                          style: TextStyle(color: Colors.white, fontSize: 30),
+                        ),
+                      );
+                    }
+                    return ListTile(
+                      leading: getLeadingIcon(
+                        context,
+                        snapshot.data![index].uniqueId,
+                      ),
+                      title: Text(
+                        snapshot.data![index].data!['barcode'].toString(),
+                      ),
+                      subtitle: Text(snapshot.data![index].data!['processName']
+                          .toString()),
+                      trailing: Wrap(
+                        children: <Widget>[
+                          IconButton(
+                              onPressed: () async {
+                                String status = 'Retrying';
+                                try {
+                                  TaskManager()
+                                      .runTask(snapshot.data![index].uniqueId);
+                                } catch (e) {
+                                  status = 'Error: $e';
+                                }
+                                final SnackBar snackBar = SnackBar(
+                                  content: Text(status),
+                                  duration: const Duration(seconds: 3),
+                                );
+                                if (!mounted) {
+                                  return;
+                                }
+                                ScaffoldMessenger.of(context)
+                                    .showSnackBar(snackBar);
+                                setState(() {});
+                              },
+                              icon: const Icon(Icons.refresh)),
+                          IconButton(
+                              onPressed: () async {
+                                String status = 'Cancelled';
```

Comment:
To prevent one useless allocation, make this final.
write 'cancelled' after cancelTask